src/errorpage.cc

```diff
@@ -1401,21 +1397,19 @@ SBuf
 ErrorState::buildBody()
 {
     assert(page_id > ERR_NONE && page_id < error_page_count);
+    errLanguage.reset();
 
 #if USE_ERR_LOCALES
     /** error_directory option in squid.conf overrides translations.
      * Custom errors are always found either in error_directory or the templates directory.
      * Otherwise locate the Accept-Language header
      */
     if (!Config.errorDirectory && page_id < ERR_MAX) {
-        if (err_language && err_language != Config.errorDefaultLanguage)
-            safe_free(err_language);
-
         ErrorPageFile localeTmpl(err_type_str[page_id], static_cast<err_type>(page_id));
         if (localeTmpl.loadFor(request.getRaw())) {
             inputLocation = localeTmpl.filename;
-            assert(localeTmpl.language());
-            err_language = xstrdup(localeTmpl.language());
+            Assure(localeTmpl.errLanguage());
```

Comment:
The replaced assertion was not checking anything useful because localeTmpl.language() returned a termedBuf() result which is never nil. I suspect the assertion was actually trying to verify that the language is known/set in this context (because loadFor() has succeeded). The new Assure() does that instead. I hope I got it right.

This comment does not request any PR changes.

Reply:
> IIRC, it was an assert specifically because it was paranoia-check for future code changes producing a nil pointer. 

Sounds good. Your recollection matches (or at least does not contradict) my [suspicions](https://github.com/squid-cache/squid/pull/2144/files#r2294008942). 

> The test can go away completely now that we are no longer dealing with char *.

PR assertion should stay because we are still dealing with a pointer-like return result: localeTmpl.errLanguage() may return nil (i.e. std::nullopt) if future code changes violate the invariant I discuss in my [comment](https://github.com/squid-cache/squid/pull/2144/files#r2294008942) annotating this code.